src/services/EdFi.DataManagementService.Api/Infrastructure/LoggingMiddleware.cs

```diff
@@ -26,11 +26,19 @@ public async Task Invoke(HttpContext context, ILogger<LoggingMiddleware> logger)
         }
         catch (Exception ex)
         {
-            var response = context.Response;
-            response.ContentType = "application/json";
             logger.LogError(JsonSerializer.Serialize(new { message = "An uncaught error has occurred", error = new { ex.Message, ex.StackTrace }, traceId = context.TraceIdentifier }));
-            response.StatusCode = (int) HttpStatusCode.InternalServerError;
-            await response.WriteAsync(JsonSerializer.Serialize(new { message = "The server encountered an unexpected condition that prevented it from fulfilling the request.", traceId = context.TraceIdentifier }));
+
+            var response = context.Response;
+            if (!response.HasStarted)
```

Comment:
InvalidOperationException gets thrown if a response has started elsewhere and a middleware tries to modify it. This was happening when I was returning 204 with an empty string body, causing an exception because ASP.NET core only accepts a null body with 204, but after the response had started. The InvalidOperationException was obscuring the actual error.